sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamSuite.scala

```diff
@@ -684,6 +685,26 @@ class StreamSuite extends StreamTest {
     assert(query.exception.isEmpty)
   }
 
+  test("SPARK-44044: non-time-window") {
+    val inputData = MemoryStream[(Int, Int)]
+    val e = intercept[AnalysisException] {
+      val agg = inputData
+        .toDF()
+        .selectExpr("CAST(_1 AS timestamp) AS col1", "_2 AS col2")
+        .withWatermark("col1", "10 seconds")
+        .withColumn("rn_col", row_number().over(Window
+          .partitionBy("col1")
+          .orderBy(col("col2"))))
+        .select("rn_col", "col1", "col2")
+        .writeStream
+        .format("console")
+        .start()
+    }
+  assert(e.getMessage.contains(
+    "Unsupported window function found in column 'row_number() AS rn_col'"))
```

Comment:
Interesting...should the error message include `OVER` spec? 🤔 It might not necessarily be related to this change but an issue with window aggregation in Spark SQL in general.

Reply:
If we add `OVER`, it looks like this:

```
Unsupported window function found in column 'row_number() OVER (PARTITION BY col1 ORDER BY col2 ASC NULLS FIRST ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) as column rn_col'. Structured streaming only supports time-window aggregation using the `window` function.
```

I don't have a strong opinion on this but I didn't include it because I personally feel that I'm a little bit overwhelmed by less relevant information. What's your opinion on this?